### PR TITLE
Spectator camera prototype

### PR DESCRIPTION
**Pull request by howard-stearns**
Creates a box "camera" in front of you, and turns on the second rendering path, in which:
- The frame buffer is sized (when you turn it on) as the desktop size.
- The view frustum is positioned and oriented from the box.

Note: does not yet attempt to force the second rendering path to non-stereo.

### CODE REVIEW

**Comment by zfox23** [comment]
General comment after spending five seconds looking at the code: I'm a little concerned about naming variables, files, etc with the word "Selfie". This is explicitly _not_ a selfie cam. It's a "secondary" camera, or maybe a "spectator" camera. I am concerned that the word "selfie" will trip up devs looking at the code or APIs in the future.

**Comment by marko8904** [comment]
github still occasionally triggers builds on the old cluster. This PR resulted in a duplicate. Please ignore the incoming "All checks failed" message. The real build will update the status within the next 30min. It seems that github still has some stale data on its servers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: af98b3bd4c5aa8881419b1e9eebadd9838e7d4fb available at: https://gist.github.com/61b711e06cdc037ec5ad94fe041fa3e2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10511-af98b3b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10511-af98b3b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/38/HighFidelity-Beta-PR10511-af98b3bd4c5aa8881419b1e9eebadd9838e7d4fb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/38/HighFidelity-Beta-PR10511-af98b3bd4c5aa8881419b1e9eebadd9838e7d4fb.dmg

**Comment by howard-stearns** [comment]
Quite right zfox23. The purpose of this build was to bring things forward for samcake, while retaining familiarity with stuff he prototyped.  When we start on this for real we will rename, and also refactor a bit, per "fix hacks on code" item in technical scope section of the project doc.

**Comment by zfox23** [comment]
Understood, @howard-stearns. Thanks!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/425/HighFidelity-Beta-PR10511-4ccfef97025a842d6b42aeddef6ca40cbdbe0dda.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/425/HighFidelity-Beta-PR10511-4ccfef97025a842d6b42aeddef6ca40cbdbe0dda.exe

File: scripts/developer/utilities/render/debugSelfie.js [hunk 1]
```diff
@@ -1,41 +1,70 @@
...
-   	var orientation = MyAvatar.orientation;
-	orientation = Quat.safeEulerAngles(orientation);
-	orientation.x = 0;
-	orientation = Quat.fromVec3Degrees(orientation);
-	var root = Vec3.sum(MyAvatar.position, Vec3.multiply(5, Quat.getForward(orientation)));
+var viewFinderOverlay, camera = Entities.addEntity({
+    type: 'Box',
+    dimensions: {x: 0.4, y: 0.2, z: 0.4},
+    userData: '{"grabbableKey":{"grabbable":true}}',
+    dynamic: isDynamic ? 1 : 0,
```
Comment (davidkelly) [inline scripts/developer/utilities/render/debugSelfie.js]: I changed this to just use the boolean, and it works fine.

File: scripts/developer/utilities/render/debugSelfie.js [hunk 2]
```diff
@@ -1,41 +1,70 @@
...
-                });
+var config = Render.getConfig("SelfieFrame");
+var config2 = Render.getConfig("BeginSelfie");
+function updateRenderFromCamera() {
+    var cameraData = Entities.getEntityProperties(camera, ['position', 'rotation']);
+    // FIXME: don't muck with config if properties haven't changed.
+    config2.position = cameraData.position;
+    config2.orientation = cameraData.rotation;
+    // BUG: image3d overlays don't retain their locations properly when parented to a dynamic object
+    if (isDynamic) {
```
Comment (davidkelly) [inline scripts/developer/utilities/render/debugSelfie.js]: Also after I just use bool above, I commented this out as it (_seems_ to be) unnecessary now.

**Comment by howard-stearns** [comment]
Following gentle review from zfox23, davidkelly, and samcake, we're calling this approved, with caveats that are listed for followup on our trello board.

**Pull request merged**